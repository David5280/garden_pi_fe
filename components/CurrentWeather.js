import React from 'react';
import { View, Text, StyleSheet, Dimensions } from 'react-native';


const CurrentWeather = ({ 
  weatherIcon, 
  temperature, 
  precipitaiton, 
  humidity, 
  wind 
  }) => {
    return (
      <View style={styles.currentWeatherContainer}>
      <View>
        {weatherIcon}
      </View>
      <View>
        <Text style={styles.currentWeatherTemp}> {Math.round(temperature)}Â°F</Text>
      </View>
      <View>
        <Text style={styles.label}>Chance of Rain: 
            <Text style={styles.bold}>
                {precipitaiton}%
            </Text>
        </Text>
        <Text style={styles.label}>Humidity: 
          <Text style={styles.bold}>
            {Math.round(humidity * 100)}%
          </Text>
          </Text>
          <Text style={styles.label}>Wind: 
            <Text style={styles.bold}>
              {Math.round(wind)} mph
            </Text>
          </Text>
        </View>
      </View>
    )
  }

const styles = StyleSheet.create({
  currentWeatherContainer: {
    alignItems: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 10,
    padding: 20,
    width: Dimensions.get('window').width * .85,
  },
  currentWeatherTemp: {
    fontSize: 35,
    fontWeight: 'bold'
  },
  icon: {
    width: 200,
    height: 80,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  bold: {
    fontSize: 15,
    fontWeight: 'bold'
  },
  label: {
    fontSize: 13,
    marginTop: 5,
  },
})

export { CurrentWeather };