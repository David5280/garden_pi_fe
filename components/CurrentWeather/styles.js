import { StyleSheet, Dimensions } from 'react-native';

export default StyleSheet.create({
  currentWeatherContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 10,
    padding: 20,
    width: Dimensions.get('window').width * .85,
  },
  currentWeatherTemp: {
    fontSize: 35,
    fontWeight: 'bold'
  },
  icon: {
    alignItems: 'center',
    flexDirection: 'row',
    height: 80,
    justifyContent: 'center',
    width: 200,
  },
  bold: {
    fontSize: 15,
    fontWeight: 'bold'
  },
  label: {
    fontSize: 13,
    marginTop: 5,
  },
});