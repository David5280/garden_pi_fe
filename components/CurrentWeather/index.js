import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles';


const CurrentWeather = ({ 
  weatherIcon, 
  temperature, 
  precipitaiton, 
  humidity, 
  wind 
  }) => {
    return (
      <View style={styles.currentWeatherContainer}>
      <View>
        {weatherIcon}
      </View>
      <View>
        <Text style={styles.currentWeatherTemp}> {Math.round(temperature)}Â°F</Text>
      </View>
      <View>
        <Text style={styles.label}>Chance of Rain: 
            <Text style={styles.bold}>
                {precipitaiton}%
            </Text>
        </Text>
        <Text style={styles.label}>Humidity: 
          <Text style={styles.bold}>
            {Math.round(humidity * 100)}%
          </Text>
          </Text>
          <Text style={styles.label}>Wind: 
            <Text style={styles.bold}>
              {Math.round(wind)} mph
            </Text>
          </Text>
        </View>
      </View>
    )
  };

export { CurrentWeather };