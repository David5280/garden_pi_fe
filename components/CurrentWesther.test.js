import React from 'react';
import { CurrentWeather } from './CurrentWeather';
import 'react-native';
import renderer from 'react-test-renderer';


test('renders correctly', () => {
  const tree = renderer.create(<CurrentWeather />).toJSON();
  expect(tree).toMatchSnapshot();
});