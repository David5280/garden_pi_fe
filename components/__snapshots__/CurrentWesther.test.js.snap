// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 10,
      "padding": 20,
      "width": 637.5,
    }
  }
>
  <View />
  <View>
    <Text
      style={
        Object {
          "fontSize": 35,
          "fontWeight": "bold",
        }
      }
    >
       
      NaN
      Â°F
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 13,
          "marginTop": 5,
        }
      }
    >
      Chance of Rain:
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        %
      </Text>
    </Text>
    <Text
      style={
        Object {
          "fontSize": 13,
          "marginTop": 5,
        }
      }
    >
      Humidity:
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        NaN
        %
      </Text>
    </Text>
    <Text
      style={
        Object {
          "fontSize": 13,
          "marginTop": 5,
        }
      }
    >
      Wind:
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        NaN
         mph
      </Text>
    </Text>
  </View>
</View>
`;
